jerry-core/ecma/base/ecma-globals.h

```diff
@@ -58,7 +58,8 @@ typedef enum
   ECMA_TYPE_SIMPLE, /**< simple value */
   ECMA_TYPE_NUMBER, /**< 64-bit integer */
   ECMA_TYPE_STRING, /**< pointer to description of a string */
-  ECMA_TYPE_OBJECT /**< pointer to description of an object */
+  ECMA_TYPE_OBJECT, /**< pointer to description of an object */
+  ECMA_TYPE___MAX = ECMA_TYPE_OBJECT /** highest value for ecma types */
```

Comment:
Just a clarification comment: `__MAX` is the highest possible value, `__COUNT` is the number of possible values. If an enum is numbered from 0, as usual, then `__COUNT == __MAX + 1`. However, if enum labels have "arbitrarily" set values, then either `__MAX` or `__COUNT` may make no sense to use.